server/dining/balance.py

```diff
@@ -120,6 +122,8 @@ def get_average_balances_by_day():
             })
 
         df = pd.DataFrame(balance_array).groupby('timestamp').agg(lambda x: x.mean()).reset_index()
+        if startDate and endDate:
+            df = df[(df['timestamp'] > startDate) & (df['timestamp'] < endDate)]
```

Comment:
Could we do the date filtering in the actual SQL query instead? It will be a lot faster and not require us to download and calculate unnecessary information just to filter it later.